### DISCUSSION TITLE
hyperwalk indicator for osu catch.

### DISCUSSION THREAD

**Discussion by ianpep** [ppy/osu#30044] [Ideas]
as stated in the title. a way to skin in a separate color for a hyperwalk would be very useful to newer players, since it is never explicitly taught and can be hard to judge on first glance without learning the map before hand.

**Comment by peppy** [comment]
what is a hyperwalk? lol.

![Safari 2024-10-03 at 05 20 07](https://github.com/user-attachments/assets/419f434c-2b17-4d31-bb12-51b0449c2472)

**Reply by cihe13375** [reply]
I think they meant the dash, i.e. the next fruit could only be caught with shift pressed but not far enough to trigger hyperdash (red fruit).
As someone previously tried ctb for like 10~20 hours (can barely pass platter) I agree this would be useful, though it's probably better as a mod than a skin setting?

**Reply by peppy** [reply]
I feel like learning which to use is part of the skill?

**Reply by cihe13375** [reply]
It could also be interpreted in that way, it's more like a matter of the learning curve. Not really a good comparison, but it's somewhat like saying "distance snap is unnecessary because it's a skill to read the rhythm from approach circles" (in ctb 1/1 walk and 1/2 dash can have similar horizontal distances). Maybe also need input from real ctb players.

**Reply by aiyern** [reply]
What cihe described is commonly call "Edge dash". 

"Hyperwalk" is when the bonus speed from a hyperdash is big enough that you can walk through the rest of the pattern.

I don't think this matter too much since "Hyperwalk" basically only exists in high difficulty loved maps and "Edge dash" only exists either as a gimmick or completely unintended.

**Reply by cihe13375** [reply]
oh indeed there are discussions about this term in the forum posts. Sorry for causing confusion :(

**Reply by ianpep** [reply]
> I don't think this matter too much since "Hyperwalk" basically only exists in high difficulty loved maps and "Edge dash" only exists either as a gimmick or completely unintended.

these note types *do* exist in quite a few higher difficulty maps, even if its not immediately obvious. The one example that comes immediately to my mind is the ending of https://osu.ppy.sh/beatmapsets/1178513#fruits/2457743, but the pattern still exists in quite a few more maps. Loved maps are primarily where you can find them in mass, but they do exist in the ranked section, where they are mixed in with just regular dashes and can be hard to differentiate.

Edge dashes are a completely different pattern, which is usually well indicated in the ranked maps that use them, and are quite common in converts.
